### PR TITLE
LPS-135136 Create set method for account type

### PR DESCRIPTION
**Pull request by erickmonteiroo**

### CODE REVIEW

**Comment by erickmonteiroo** [comment]
ci:stop

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -58,6 +58,11 @@ components:
                     type: integer
                 status:
                     type: integer
+                type:
+                    # @review
+                    description:
+                        The account's type. This property will be ignored during PATCH and PUT requests.
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml:64]: Now that I re-read this, I think it's not accurate anymore. PUT requests with an externalReferenceCode will function as an upsert, so an Account could be created and the type could be read.

Let's change this to say: "The account's type. This property will be read during Account creation, but ignored during updates."

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java [hunk 2]
```diff
@@ -498,6 +508,22 @@ private void _testGetAccountsPage(
...
+				com.liferay.headless.admin.user.dto.v1_0.Account.Type.create(
+					StringUtil.randomString()));
+
+			Assert.fail();
+		}
+		catch (Exception exception) {
+			Assert.assertTrue(exception instanceof IllegalArgumentException);
+		}
+	}
+
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java:526]: I don't think we need this case anymore.  The enum type should handle this aready.

**Review by drewbrokke — CHANGES_REQUESTED** [review]

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Review by drewbrokke — APPROVED** [review]

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**